packages/@webex/plugin-meetings/src/common/collection.ts

```diff
@@ -41,7 +43,7 @@ export default class Collection {
    * @public
    * @memberof Collection
    */
-  get(id) {
+  public get(id: string) {
```

Comment:
We're adding types, but not removing the types from jsdoc, so there is duplication and which one is the source of truth now when there is a mismatch? We can now have a different type in jsdoc, different declared in typescript and because of the "// @ts-nocheck" line everything will be ignored, see this example screenshot:
![image](https://user-images.githubusercontent.com/7473332/205131306-25a947d1-a0b1-4425-b990-a64f81aa7aba.png)

I've just taken a random piece of code (constructor of MuteState class) and modified the type of the `meeting` argument in the code and also modified the jsdoc, so there is a complete mismatch: jsdoc says it's a number, typescript declaration says it's an object with 2 fields "a" and "b" and the code is actually accessing fields "remoteMuted" and "unmuteAllowed" - so it's all completely wrong and yet I have no warnings or errors in vscode, I can build the code fine, I've also run "yarn lint" and everything passes, there is no indication anywhere that anything is wrong.

Reply:
Fixed.

Reply:
ok so we no longer have ts-nocheck, but what about the jsdocs? I've just done a quick check - I've checked out this PR and modified one of the types in jsdoc to be wrong and I don't see any error in vscode. Is there maybe some tsconfig option to check if types in jsdoc match the typescript code? Which type is used in the generated docs? the one from jsdoc (which potentially can be incorrect) or from typescript? I think we should remove the types from jsdocs and move to tsdoc or typedoc or something like that, right? What's the plan for this?

Reply:
Sounds right, We can create a ticket for this.

CC - @arun3528

Reply:
JIRA Ticket - https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-393492